docs/api.rst

```diff
@@ -189,13 +189,70 @@ It allows you to maintain and update a "state" (aka feedback, aka seed for the n
   	return { seed: values, value: avg }
   }, [], stream)
 
-zipArrayValues :: ((a, b) -> c) -> [a] -> Stream b -> Stream c
+.. _zipArrayValues 
 
-withArrayValues :: [a] -> Stream b -> Stream a
+zipArrayValues
+^^^^^^^^^^^^^^
 
-chain :: (a -> Stream b) -> Stream a -> Stream b
+.. code-block:: haskell
+
+  zipArrayValues :: ((a, b) -> c) -> [a] -> Stream b -> Stream c
+
+Creates a new stream by applying a function ``((a, b) -> c)`` to the latest value of ``Stream b`` with an increasing index of ``[a]`` until ``Stream b`` ends or there are no more values in ``[a]``.::
+
+  stream:                             --10---10---10---10---10--->
+  array:                              [ 1, 2, 3 ]
+  zipArrayValues(add, array, stream): --11---12---13|
+
+.. _withArrayValues
+
+withArrayValues
+^^^^^^^^^^^^^^^
+
+.. code-block:: haskell
+
+  withArrayValues :: [a] -> Stream b -> Stream a
+
+Creates a new stream by associating the time of event occurence from ``stream`` with the values of an array. The new stream 
+will end either when there are no more values in the array or the underlying stream has ended.::
```

Comment:
Depending on where we end up with `zipArrayValues`, we can synchronize the style of `withArrayValues` to match (like you did here :+1:)